docs/components/input-controller/_index.md

```diff
@@ -194,51 +194,45 @@ For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/c
 ```go {class="line-numbers linkable-line-numbers"}
 
 // Define a function that handles the controller.
-func handleController(controller input.Controller) {
-
-    // Define a function to handle pressing the Start Menu Button "ButtonStart" on your controller, logging the start time.
+func handleController(ctx context.Context, logger logging.Logger, controller input.Controller) error {
+    // Define a function to handle pressing the Start Menu button, "ButtonStart", on your controller and logging the start time
     printStartTime := func(ctx context.Context, event input.Event) {
         logger.Info("Start Menu Button was pressed at this time: %v", event.Time)
     }
 
     // Define the EventType "ButtonPress" to serve as the trigger for printStartTime.
-    triggers := [1]input.EventType{input.ButtonPress}
+    triggers := []input.EventType{input.ButtonPress}
 
     // Get the controller's Controls.
-    controls, err := controller.Controls(context.Background(), nil)
+    controls, err := controller.Controls(ctx, nil)
 
     // If the "ButtonStart" Control is found, register the function printStartTime to fire when "ButtonStart" has the event "ButtonPress" occur.
-    if slices.Contains(controls, Control.ButtonStart) {
-        err := controller.RegisterControlCallback(context.Background(), Control: input.ButtonStart, triggers, printStartTime, nil)
-    }
-     else {
-        logger.Fatalf("Oops! Couldn't find the start button control! Is your controller connected?")
+    if !slices.Contains(controls, input.ButtonStart) {
+        return errors.New("button `ButtonStart` not found; controller may be disconnected")
     }
+    return controller
```

Comment:
```suggestion
    return controller.RegisterControlCallback(context.Background(), Control: input.ButtonStart, triggers, printStartTime, nil)
```
I don't think you're actually registering the controller callback here; you need to call `RegisterControlCallback` and return its result (an `error` type).